xonsh/tools.py

```diff
@@ -92,6 +93,85 @@ def __init__(self, returncode, command, output=None, stderr=None,
         self.completed_command = completed_command
 
 
+class EnvPath(MutableSequence):
```

Comment:
This would create a cycle of dependencies, since `str_to_env_path` (which is in xonsh.tools) should return an `EnvPath` object.